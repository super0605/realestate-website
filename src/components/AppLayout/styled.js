import styled from "styled-components";
import { Layout } from "antd";

const { Content } = Layout;

export const LayoutContainer = styled(Layout)`
  min-height: 100vh;
`;

export const MainWrapContainer = styled(Content)`
  background: #f8f9fa;
  z-index: 1;
`;

export const MainContainer = styled.div`
  margin-top: 148px;
`;

export const Wrapper = styled.div`
  height: 100%;
  width: 100%;
  position: relative;
  & .feeprofile-new-modal {
    position: absolute;
  }
`;
