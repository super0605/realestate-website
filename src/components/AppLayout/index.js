import React, { useState } from "react";
import PropTypes from "prop-types";
import { Layout } from "antd";
import Header from "../Header";
import Sidebar from "../Sidebar";
import { LayoutContainer, MainWrapContainer, MainContainer, Wrapper } from "./styled";

const AppLayout = ({ children, type }) => {
  const [showSidebar, setShowSidebar] = useState(false);

  const handleCloseSidebar = () => {
    setShowSidebar(false);
  };

  return (
    <LayoutContainer>
      <Layout>
        <MainWrapContainer>
          <Header handleSidebar={setShowSidebar} />
          <MainContainer type={type}>
            <Wrapper>{children}</Wrapper>
          </MainContainer>
        </MainWrapContainer>
      </Layout>
      <Sidebar isVisible={showSidebar} handleClose={handleCloseSidebar} />
    </LayoutContainer>
  );
};

AppLayout.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(["default", "no_sidebar"]),
};

AppLayout.defaultProps = {
  type: "default",
};

export default AppLayout;
