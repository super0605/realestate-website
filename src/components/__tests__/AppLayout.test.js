import * as React from "react";
import { mount, shallow } from "enzyme";
import AppLayout from "../AppLayout";

describe("AppLayout Component", () => {
  let props;
  let mountedAppLayout;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedAppLayout) {
        mountedAppLayout = bShallow
        ? shallow(<AppLayout {...props} />)
        : mount(<AppLayout {...props} />);
    }
    return mountedAppLayout;
  };

  beforeEach(() => {
    props = {
      children: "<div>Children element</div>"
    }
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
