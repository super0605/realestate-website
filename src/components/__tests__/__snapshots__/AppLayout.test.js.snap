// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Component should render without problems 1`] = `
<Styled(Layout)>
  <Layout>
    <ContextProvider
      value={
        Object {
          "getContainer": [Function],
        }
      }
    >
      <Styled(Content)>
        <HeaderComponent />
        <styled.div
          type="default"
        >
          <styled.div>
            &lt;div&gt;Children element&lt;/div&gt;
          </styled.div>
        </styled.div>
      </Styled(Content)>
    </ContextProvider>
  </Layout>
</Styled(Layout)>
`;
